### PR TITLE
Add option for custom build file name

### LINKED ISSUE #111: Add support for using BUILD.bazel instead of BUILD
Bazel has added the option to call build files `BUILD.bazel` instead of `BUILD`, which is less likely to conflict with other files or directories. bazel-deps always creates files called `BUILD`, which leads to two issues:

- inconsistency between generated files and the rest of the tree
- creation of an empty file called `BUILD` at the workspace root, which will be ignored by Bazel if `BUILD.bazel` is present

It would be nice if bazel-deps had an option to create files called `BUILD.bazel` instead.

### PR DESCRIPTION
**Pull request by mjduijn**
Allows for specifying a custom filename as the filename for generated build files.

Resolves  #111

### CODE REVIEW

File: src/scala/com/github/johnynek/bazel_deps/Writer.scala [hunk 1]
```diff
@@ -156,8 +154,6 @@ object Writer {
         |    ]
         |
         |def maven_dependencies(callback = jar_artifact_callback):
-        |    for hash in list_dependencies():
```
Comment (johnynek) [inline src/scala/com/github/johnynek/bazel_deps/Writer.scala]: Why is this removed?

Comment (mjduijn) [reply src/scala/com/github/johnynek/bazel_deps/Writer.scala]: That shouldn't have been changed.
My editor started autoformatting edited files, in the process of undoing this I must have deleted these lines.
